pkgs/servers/typedb/default.nix

```diff
@@ -0,0 +1,130 @@
+        sha256 = "00wvkiahmrys32ckp6r59bbrzfd7jzsphniyjdj4p9bh7k339gn2";
+      };
+      dir = "typedb-all-linux-${typedbVersion}";
+    };
+    x86_64-darwin = {
+      src = fetchzip {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-mac-2.1.1.zip";
+        sha256 = "16hlfy6kh2rnvcralz206q13mghb0rv8wazpg6q3h324p5rdys54";
+      };
+      dir = "typedb-all-mac-${typedbVersion}";
+    };
+    x86_64-windows = {
+      src = fetchzip {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-windows-2.1.1.zip";
+        sha256 = "0vd66gfshkg697z07nhy957mwqzlli4r4pmn67hx58n9mkg024kq";
+      };
+      dir = "typedb-all-windows-${typedbVersion}";
+    };
+  };
+
+  currentSystem = systems.${system} or (throw "unsupported system ${system}");
+  typedbDir = currentSystem.dir;
+  srcFolder = currentSystem.src;
+  javaPatch = ''
+    20c20
+    < JAVA_BIN=java
+    ---
+    > JAVA_BIN=${openjdk}/bin/java
+    32c32
+    <     which "$'' + "{JAVA_BIN}\" > /dev/null\n" +
+  "---\n" +
+  ">     ${which}/bin/which \"$" + "{JAVA_BIN}\" > /dev/null";
+
+  typedb-properties-patch = ''
+    19c19
+    < server.data=server/data/
+    ---
+    > server.data=${server-data}
+    22c22
+    < server.logs=server/logs/
+    ---
+    > server.logs=${server-logs}
+    24c24
+    < server.port=1729
+    ---
+    > server.port=${server-port}
+  '';
+  typedb-wrapper-patch = ''
+    3c3
+    <   TYPEDB_SERVER_DATA="__SERVER_DATA__"
+    ---
+    >   TYPEDB_SERVER_DATA="${server-data}"
+    8c8
+    <   TYPEDB_SERVER_LOGS="__SERVER_LOGS__"
+    ---
+    >   TYPEDB_SERVER_LOGS="${server-logs}"
+    13c13
+    <   TYPEDB_SERVER_PORT="__SERVER_PORT__"
+    ---
+    >   TYPEDB_SERVER_PORT="${server-port}"
+  '';
+  typedbWrapper = ./typedbWrapper.sh;
+
+in
+stdenv.mkDerivation rec {
+  pname = "typedb";
+  version = typedbVersion;
+
+  src = srcFolder;
+
+  dontBuild = true;
+
+  /*
+    in preparation for the coming change of properties file parsing
+    allowing for environment variable reading :
+    > echo "${typedb-properties-patch}" > typedb-properties.patch
+    > patch ./server/conf/typedb.properties typedb-properties.patch
+  */
+  installPhase = ''
+    #patch before install
+    echo '${javaPatch}' > typedb_java.patch
+    echo '${typedb-wrapper-patch}' > typedb_wrapper.patch
+    patch ./typedb typedb_java.patch
+
+    mkdir $out
+    cp -r ./* $out
+    mkdir $out/bin
+
+    cat ${typedbWrapper} > $out/bin/typedb
+    echo "$out/typedb server --data=\$TYPEDB_SERVER_DATA --logs=\$TYPEDB_SERVER_LOGS --port=\$TYPEDB_SERVER_PORT \"\$@\"" >> $out/bin/typedb
+    patch $out/bin/typedb typedb_wrapper.patch
+
+    chmod u+x $out/bin/typedb
+    rm typedb_java.patch
+    rm typedb_wrapper.patch
+  '';
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    mkdir "$out/server_data"
+    mkdir "$out/server_log"
```

Comment:
This will end up in the package, but shouldn't.
You could test it in a separate derivation instead, via `passthru.tests`.
See https://nixos.org/manual/nixpkgs/stable/#sec-package-tests and https://nixos.org/manual/nixpkgs/stable/#var-meta-tests

Reply:
Well, you remove it later, but we can't know for sure this is all that gets written.

Reply:
alright; thanks for the appropriate docs

Reply:
moved all `installCheckPhase` tests to the `passthru` test-suite